api/app/controllers/review.controller.ts

```diff
@@ -581,3 +432,99 @@ export const resetReviews = async (req: Request, res: Response) => {
     res.status(500).send(`Error resetting review: ${err}`);
   }
 };
+
+
+
+// utility functions.
+
+const buildCondition = (
+  status: string | undefined,
+  transcriptId: number | undefined,
+  userId: number | undefined,
+  mergedAt: string | undefined,
+  userSearch: string | undefined
+) => {
+  const condition: { [key: string | number]: any } = {};
+  const userCondition: { [Op.or]?: { email?: { [Op.iLike]: string }; githubUsername?: { [Op.iLike]: string } }[] } = {};
+
+  if (status) {
+    const currentTime = new Date().getTime();
+    switch (status) {
+      case QUERY_REVIEW_STATUS.ACTIVE:
+        const activeCondition = buildIsActiveCondition(currentTime);
+        condition[Op.and as unknown as keyof typeof Op] = activeCondition;
+        break;
+
+      case 'expired':
+        const expiredCondition = buildIsInActiveCondition(currentTime);
+        condition[Op.and as unknown as keyof typeof Op] = expiredCondition;
+        break;
+
+      case QUERY_REVIEW_STATUS.PENDING:
+        const pendingCondition = buildIsPendingCondition();
+        condition[Op.and as unknown as keyof typeof Op] = pendingCondition;
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  if (mergedAt) {
+    const date = new Date(mergedAt);
+    const startOfDay = new Date(date.getFullYear(), date.getMonth(), date.getDate());
+    const endOfDay = new Date(
+      date.getFullYear(),
+      date.getMonth(),
+      date.getDate(),
+      HOUR_END_OF_DAY,
+      MINUTE_END_OF_DAY,
+      SECOND_END_OF_DAY,
+      MILLISECOND_END_OF_DAY
+    );
+
+    condition.mergedAt = {
+      [Op.gte]: startOfDay,
+      [Op.lte]: endOfDay,
+    };
+  }
+
+  if (transcriptId) {
+    condition.transcriptId = { [Op.eq]: transcriptId };
+  }
+
+  if (userId) {
+    condition.userId = { [Op.eq]: userId };
+  }
+
+  if (userSearch) {
+    const searchCondition = { [Op.iLike]: `%${userSearch.toLowerCase()}%` };
+    userCondition[Op.or] = [
+      { email: searchCondition },
+      { githubUsername: searchCondition },
+    ];
+  }
+
+  return { condition, userCondition };
+};
+
+const buildResponse = (
```

Comment:
lets rename this to `buildReviewResponse` since this is specific to reviews. you can also move it to the utils folder
```suggestion
const buildReviewResponse = (
```